### ISSUE TITLE
APITool slows down startup drastically 

### ISSUE THREAD

**Issue by aclowkey** [jooby-project/jooby#1221]
The API tool is very good and useful for documentation, but is rarely accessed. Yet causes very slow start ups.

With ApiTool: 

> [2018-07-23 15:09:40,081]-[main] INFO  c.z.s.App - [local@netty]: Server started in 34232ms

Without ApiTool:

> [2018-07-23 15:10:09,926]-[main] INFO  c.z.s.App - [local@netty]: Server started in 2860ms

We looked into running the maven apitool but we can't configure it, and also has some issues.

```
com.fasterxml.jackson.databind.JsonMappingException: Index: 0, Size: 0 (through reference chain: java.util.ArrayList[5]->org.jooby.apitool.RouteMethod["response"]->org.jooby.apitool.RouteResponse["type"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:391)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:351)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1711)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:290)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:136)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2985)
	at org.jooby.internal.apitool.BytecodeRouteParser.read(BytecodeRouteParser.java:364)
	at org.jooby.internal.apitool.BytecodeRouteParser.lambdas(BytecodeRouteParser.java:595)
	at org.jooby.internal.apitool.BytecodeRouteParser.parse(BytecodeRouteParser.java:397)
	at org.jooby.apitool.ApiParser.parse(ApiParser.java:295)
	at org.jooby.apitool.ApiParser.parseFully(ApiParser.java:310)
	at org.jooby.internal.apitool.APIProvider.<init>(APIProvider.java:225)
	at org.jooby.internal.apitool.APIProvider$$FastClassByGuice$$9b325895.newInstance(<generated>)
	at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:60)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:148)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:211)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:182)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
	at com.google.inject.Guice.createInjector(Guice.java:87)
	at com.google.inject.Guice.createInjector(Guice.java:78)
	at org.jooby.Jooby.lambda$new$0(Jooby.java:885)
	at org.jooby.Jooby.bootstrap(Jooby.java:2996)
	at org.jooby.Jooby.start(Jooby.java:2197)
	at org.jooby.Jooby.start(Jooby.java:2178)
	at org.jooby.Jooby.run(Jooby.java:2107)
	at co.zeekit.server.App.main(App.java:296)
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:657)
	at java.util.ArrayList.get(ArrayList.java:433)
	at org.jooby.internal.apitool.TypeJsonDeserializer.deserialize(TypeJsonDeserializer.java:226)
	at org.jooby.internal.apitool.TypeJsonDeserializer.deserialize(TypeJsonDeserializer.java:217)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:136)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	... 35 common frames omitted

Process finished with exit code 130 (interrupted by signal 2: SIGINT)
```

**Comment by aclowkey** [comment]
Maybe the swagger.json should be generated lazily

**Comment by jknack** [comment]
hmm is expected to be **slow** while running on development (application.env=dev)

for none `dev` environment you need to add the maven/gradle plugin: https://jooby.org/doc/apitool/#API-tool-keep-documentation

Then you need to start your application with `application.env=prod` (or anything except `dev`)

Are you using the maven/gradle build tool?

**Comment by aclowkey** [comment]
@jknack I'm adding `use(new ApiTool()..)` in the constructor
I think for it to work quickly is more critical for development due to many restarts.. 
I use maven build tool

**Comment by jknack** [comment]
wait, I missed the comment about maven plugin.

Is the maven plugin working? If not I need a way to reproduce the maven/build issue

**Comment by jknack** [comment]
Pushed a possible fix for `IndexOutOfBoundsException`.

Also, make the apiparser to work lazily (on first request). It caches the result for none `dev` env.

Are you able to try and test agains the master branch?

**Comment by aclowkey** [comment]
I'm unable to build the project.. Getting
`mvn install`
>  initializationError(org.jooby.JoobyTest)  Time elapsed: 0.001 s  <<< ERROR!
org.objenesis.ObjenesisException: java.lang.reflect.InvocationTargetException
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.IllegalAccessError: class jdk.internal.reflect.ConstructorAccessorImpl loaded by org/powermock/core/classloader/MockClassLoader cannot access jdk/internal/reflect superclass jdk.internal.reflect.MagicAccessorImpl

**Comment by jknack** [comment]
make sure you build against java 8 and don't run tests:

    mvn clean install -DskipTests=true

**Comment by jknack** [comment]
pushed another change, please try a again

**Comment by aclowkey** [comment]
> [2018-07-24 15:48:31,083]-[netty task-4-2] ERROR org.jooby.internal.apitool.BytecodeRouteParser - read of /co/zeekit/server/App.json resulted in exception
> com.fasterxml.jackson.databind.JsonMappingException:  java.lang.String (through reference chain: java.util.ArrayList[71]->org.jooby.apitool.RouteMethod["response"]->org.jooby.apitool.RouteResponse["type"])
> 	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:391)
> 	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:351)
> 	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1711)
> 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:290)
> 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
> 	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:136)
> 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
> 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
> 	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
> 	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
> 	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
> 	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013)
> 	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2985)
> 	at org.jooby.internal.apitool.BytecodeRouteParser.read(BytecodeRouteParser.java:364)
> 	at org.jooby.internal.apitool.BytecodeRouteParser.lambdas(BytecodeRouteParser.java:595)
> 	at org.jooby.internal.apitool.BytecodeRouteParser.parse(BytecodeRouteParser.java:397)
> 	at org.jooby.apitool.ApiParser.parse(ApiParser.java:295)
> 	at org.jooby.apitool.ApiParser.parseFully(ApiParser.java:310)
> 	at org.jooby.apitool.ApiTool.lambda$configure$3(ApiTool.java:797)
> 	at org.jooby.funzy.Throwing$Function.lambda$null$12(Throwing.java:1174)
> 	at org.jooby.funzy.Throwing.fn(Throwing.java:2423)
> 	at org.jooby.funzy.Throwing.access$400(Throwing.java:39)
> 	at org.jooby.funzy.Throwing$Supplier.get(Throwing.java:145)
> 	at org.jooby.funzy.Throwing.memo(Throwing.java:2537)
> 	at org.jooby.funzy.Throwing.access$1000(Throwing.java:39)
> 	at org.jooby.funzy.Throwing$Function.lambda$memoized$13(Throwing.java:1173)
> 	at org.jooby.funzy.Throwing$Function.lambda$apply$0(Throwing.java:1072)
> 	at org.jooby.funzy.Throwing.fn(Throwing.java:2423)
> 	at org.jooby.funzy.Throwing.access$400(Throwing.java:39)
> 	at org.jooby.funzy.Throwing$Function.apply(Throwing.java:1072)
> 	at org.jooby.apitool.ApiTool.lambda$swagger$8(ApiTool.java:1054)
> 	at org.jooby.Route$OneArgHandler.handle(Route.java:1817)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Handler.handle(Route.java:1761)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Before.handle(Route.java:1924)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Before.handle(Route.java:1924)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$After.handle(Route.java:2002)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$After.handle(Route.java:2002)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at co.zeekit.server.module.RelaxCorsHandler.handle(RelaxCorsHandler.java:72)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at co.zeekit.server.module.auth.BasicAuth.lambda$configure$0(BasicAuth.java:45)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Handler.handle(Route.java:1761)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Handler.handle(Route.java:1761)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Handler.handle(Route.java:1761)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.internal.HttpHandlerImpl.handle(HttpHandlerImpl.java:496)
> 	at org.jooby.internal.netty.NettyHandler.channelRead0(NettyHandler.java:274)
> 	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
> 	at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:38)
> 	at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:353)
> 	at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:66)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	at java.lang.Thread.run(Thread.java:748)
> Caused by: java.lang.ClassNotFoundException:  java.lang.String
> 	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
> 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
> 	at org.jooby.internal.apitool.BytecodeRouteParser.lambda$loadType$59(BytecodeRouteParser.java:1133)
> 	at org.jooby.funzy.Throwing.fn(Throwing.java:2423)
> 	at org.jooby.funzy.Throwing.access$400(Throwing.java:39)
> 	at org.jooby.funzy.Throwing$Supplier.get(Throwing.java:145)
> 	at java.util.Optional.orElseGet(Optional.java:267)
> 	at org.jooby.funzy.When$Value.orElseGet(When.java:86)
> 	at org.jooby.internal.apitool.BytecodeRouteParser.loadType(BytecodeRouteParser.java:1133)
> 	at org.jooby.internal.apitool.TypeJsonDeserializer.parse(TypeJsonDeserializer.java:253)
> 	at org.jooby.internal.apitool.TypeJsonDeserializer.parse(TypeJsonDeserializer.java:245)
> 	at org.jooby.internal.apitool.TypeJsonDeserializer.parse(TypeJsonDeserializer.java:245)
> 	at org.jooby.internal.apitool.TypeJsonDeserializer.deserialize(TypeJsonDeserializer.java:231)
> 	at org.jooby.internal.apitool.TypeJsonDeserializer.deserialize(TypeJsonDeserializer.java:219)
> 	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:136)
> 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
> 	... 81 common frames omitted

**Comment by jknack** [comment]
can you share the App.json file?

**Comment by aclowkey** [comment]
I can't publicly share it. I can email it to you.

**Comment by jknack** [comment]
Ok, let's do that, so I'm not guessing what is wrong.

Thanks

**Comment by jknack** [comment]
Try again please.

**Comment by jknack** [comment]
Please try again :(

I didn't get the `Array` exception, just (think) I fix the NPE

I couldn't use your App.json bc I need all the your domain types.

**Comment by aclowkey** [comment]
Still getting  `java.lang.ClassNotFoundException: Array` ..

**Comment by jknack** [comment]
Cool!

**Comment by aclowkey** [comment]
Waiting for the 1.5.1 release :)

**Comment by jknack** [comment]
Going to release 1.5.1 this weekend.

**Comment by aclowkey** [comment]
@jknack We would be happy to have the APITool working again :)

**Comment by jknack** [comment]
1.5.1 released

**Comment by aclowkey** [comment]
@jknack 
Not sure what happened since, but it's not working ..

> [2018-08-28 14:11:21,153]-[netty task-4-2] ERROR co.z.App - Error on request /swagger/swagger.json
> java.lang.IllegalArgumentException: Cannot create TypeBindings for class java.util.List with 2 type parameters: class expects 1
> 	at com.fasterxml.jackson.databind.type.TypeBindings.create(TypeBindings.java:139)
> 	at com.fasterxml.jackson.databind.type.TypeBindings.create(TypeBindings.java:98)
> 	at com.fasterxml.jackson.databind.type.TypeFactory._fromParamType(TypeFactory.java:1446)
> 	at com.fasterxml.jackson.databind.type.TypeFactory._fromAny(TypeFactory.java:1213)
> 	at com.fasterxml.jackson.databind.type.TypeFactory._fromParamType(TypeFactory.java:1444)
> 	at com.fasterxml.jackson.databind.type.TypeFactory._fromAny(TypeFactory.java:1213)
> 	at com.fasterxml.jackson.databind.type.TypeFactory.constructType(TypeFactory.java:623)
> 	at com.fasterxml.jackson.databind.ObjectMapper.constructType(ObjectMapper.java:1687)
> 	at org.jooby.internal.apitool.FriendlyTypeName.name(FriendlyTypeName.java:218)
> 	at org.jooby.internal.apitool.SwaggerBuilder.lambda$buildResponse$13(SwaggerBuilder.java:479)
> 	at java.util.Optional.orElseGet(Optional.java:267)
> 	at org.jooby.internal.apitool.SwaggerBuilder.buildResponse(SwaggerBuilder.java:479)
> 	at org.jooby.internal.apitool.SwaggerBuilder.build(SwaggerBuilder.java:432)
> 	at org.jooby.apitool.ApiTool.lambda$swagger$8(ApiTool.java:1054)
> 	at org.jooby.Route$OneArgHandler.handle(Route.java:1817)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at co.zeekit.server.utility.Metrics.lambda$configure$2(Metrics.java:58)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Handler.handle(Route.java:1761)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Before.handle(Route.java:1924)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Before.handle(Route.java:1924)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$After.handle(Route.java:2002)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$After.handle(Route.java:2002)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at co.zeekit.server.module.RelaxCorsHandler.handle(RelaxCorsHandler.java:72)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at co.zeekit.server.module.auth.BasicAuth.lambda$configure$0(BasicAuth.java:45)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Handler.handle(Route.java:1761)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.Route$Handler.handle(Route.java:1761)
> 	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
> 	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
> 	at org.jooby.Route$Chain.next(Route.java:2163)
> 	at org.jooby.internal.HttpHandlerImpl.handle(HttpHandlerImpl.java:496)
> 	at org.jooby.internal.netty.NettyHandler.channelRead0(NettyHandler.java:274)
> 	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
> 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
> 	at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:38)
> 	at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:353)
> 	at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:66)
> 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	at java.lang.Thread.run(Thread.java:748)

**Comment by aclowkey** [comment]
> java.lang.ClassNotFoundException: Array
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at io.swagger.util.ReflectionUtils.loadClassByName(ReflectionUtils.java:53)
	at io.swagger.jackson.ModelResolver.getInnerType(ModelResolver.java:916)
	at io.swagger.jackson.ModelResolver.resolve(ModelResolver.java:459)
	at io.swagger.jackson.ModelResolver.resolve(ModelResolver.java:194)
	at io.swagger.jackson.AbstractModelConverter.resolve(AbstractModelConverter.java:202)
	at io.swagger.converter.ModelConverterContextImpl.resolve(ModelConverterContextImpl.java:103)
	at io.swagger.jackson.ModelResolver.resolve(ModelResolver.java:248)
	at io.swagger.jackson.ModelResolver.resolve(ModelResolver.java:194)
	at io.swagger.jackson.AbstractModelConverter.resolve(AbstractModelConverter.java:202)
	at io.swagger.converter.ModelConverterContextImpl.resolve(ModelConverterContextImpl.java:103)
	at io.swagger.jackson.ModelResolver.resolveProperty(ModelResolver.java:170)
	at io.swagger.jackson.ModelResolver.resolveProperty(ModelResolver.java:121)
	at io.swagger.jackson.AbstractModelConverter.resolveProperty(AbstractModelConverter.java:79)
	at org.jooby.internal.apitool.SwaggerBuilder$1.resolveProperty(SwaggerBuilder.java:288)
	at io.swagger.converter.ModelConverterContextImpl.resolveProperty(ModelConverterContextImpl.java:83)
	at io.swagger.converter.ModelConverters.readAsProperty(ModelConverters.java:63)
	at io.swagger.converter.ModelConverters.readAsProperty(ModelConverters.java:57)
	at org.jooby.internal.apitool.SwaggerBuilder.lambda$build$6(SwaggerBuilder.java:387)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.jooby.internal.apitool.SwaggerBuilder.build(SwaggerBuilder.java:429)
	at org.jooby.apitool.ApiTool.lambda$swagger$8(ApiTool.java:1054)
	at org.jooby.Route$OneArgHandler.handle(Route.java:1817)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at co.zeekit.server.utility.Metrics.lambda$configure$2(Metrics.java:58)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.Route$Handler.handle(Route.java:1761)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.Route$Before.handle(Route.java:1924)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.Route$Before.handle(Route.java:1924)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.Route$After.handle(Route.java:2002)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.Route$After.handle(Route.java:2002)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at co.zeekit.server.module.RelaxCorsHandler.handle(RelaxCorsHandler.java:72)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at co.zeekit.server.module.auth.BasicAuth.lambda$configure$0(BasicAuth.java:45)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.Route$Handler.handle(Route.java:1761)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.Route$Handler.handle(Route.java:1761)
	at org.jooby.internal.RouteImpl.handle(RouteImpl.java:280)
	at org.jooby.internal.RouteChain.next(RouteChain.java:262)
	at org.jooby.Route$Chain.next(Route.java:2163)
	at org.jooby.internal.HttpHandlerImpl.handle(HttpHandlerImpl.java:496)
	at org.jooby.internal.netty.NettyHandler.channelRead0(NettyHandler.java:274)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:38)
	at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:353)
	at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:66)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

**Comment by jknack** [comment]
I'm to close it again, this is issue it is hard to follow but also has been fixed (apitool slow down startup). Please open a new issue add an example to reproduce it and paste the stacktrace.

Thanks
